src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_ssl.c

```diff
@@ -462,15 +462,18 @@ int32_t AppleCryptoNative_SslIsHostnameMatch(SSLContextRef sslContext, CFStringR
                 {
                     CFArrayRef statusCodes = CFDictionaryGetValue(CFArrayGetValueAtIndex(details,0), CFSTR("StatusCodes"));
 
-                    OSStatus status = 0;
-                    // look for first failure to keep it simple. Normally, there will be exactly one.
-                    for (int i = 0; i < CFArrayGetCount(statusCodes); i++)
+                    if (statusCodes != NULL)
```

Comment:
Should this set some unknown pOSStatus in NULL case?

Reply:
That was my initial thought as well but the code ignores these private attributes if they are not present. Hence I am treating `trust->TrustResultDetails->StatusCodes` not present the same as `trust->TrustResultDetails` not present. Essentially the result is that you will get less specific error code but still a trust failure.